packages/plugin-layout/src/LayoutEngine.ts

```diff
@@ -0,0 +1,277 @@
+import { Constructor } from '../../utils/src/utils';
+import { VNode } from '../../core/src/VNodes/VNode';
+import JWEditor from '../../core/src/JWEditor';
+import { ZoneIdentifier, ZoneNode } from './ZoneNode';
+import { FragmentNode } from '../../core/src/VNodes/FragmentNode';
+
+export type componentReference = string | VNode;
+export type DomZonePosition = 'after' | 'prepend' | 'append' | 'replace';
+
+export interface Component<N extends Constructor<VNode> = Constructor<VNode>> {
+    zone: ZoneIdentifier;
+    VNode?: N | InstanceType<N>;
+    args?: ConstructorParameters<N>;
```

Comment:
The only thing that we need when defining a component is a way to get it's VNode. 
These two properties generate clutter in the code because we have to make branching to handle them.
This could simply be avoided by providing a function that retrieve the VNode describing the Component
```typescript
getVnode() => VNode
```